test/Npgsql.Tests/MultipleHostsTests.cs

```diff
@@ -953,6 +952,27 @@ async Task Query(NpgsqlDataSource dataSource)
         }
     }
 
+    [Test]
+    [IssueLink("https://github.com/npgsql/npgsql/issues/5055")]
+    [NonParallelizable] // Disables sql rewriting
+    public async Task Multiple_hosts_with_disabled_sql_rewriting()
+    {
+        using var _ = DisableSqlRewriting();
+        PoolManager.Reset();
```

Comment:
nit: we could move this into DisableSqlRewriting

Reply:
Sure, just was thinking that having this in `MultipleHostsTests` is a bit more appropriate.

Reply:
Ah, you mean to move `PoolManager.Reset`?

Reply:
Sure. Was just saying that we probably want to clear the pool wherever we have a test doing such a global change etc.

Reply:
OK, can do. The only reason I added `PoolManager.Reset` is because I copy-pasted from `IntegrationTest` above (it was required because we were caching cluster's state globally, and now it's probably not needed anymore).